test/test_memcached_store.rb

```diff
@@ -744,6 +744,28 @@ def test_reset_does_raise_on_error
     end
   end
 
+  def test_append_raises_not_stored_on_miss
+    assert_raises Memcached::NotStored do
```

Comment:
So this should instead assert that append returns `false` and `test_append` should assert that it returns `true`.